File: elearning-platform/elearning-platform/src/main/java/com/elearning/elearning_platform/user/service/UserService.java
package com.elearning.elearning_platform.user.service;

import com.elearning.elearning_platform.user.dto.CreateUserRequest;
import com.elearning.elearning_platform.user.dto.UserResponse;

import java.util.List;

public interface UserService {
    UserResponse create(CreateUserRequest request);
    UserResponse getById(Long id);
    List<UserResponse> list();

    // Missing methods added
    UserResponse updateProfile(Long userId, com.elearning.elearning_platform.user.dto.UpdateProfileRequest request);
    UserResponse toggleStatus(Long id);
}

File: Frontend/Frontend/src/app/services/user.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { CreateUserRequest, UserResponse, UpdateProfileRequest } from '../dtos/user.dto';

@Injectable({
  providedIn: 'root'
})
export class UserService {
  private readonly apiUrl = `${environment.apiUrl}/users`;

  constructor(private http: HttpClient) {}

  create(request: CreateUserRequest): Observable<UserResponse> {
    return this.http.post<UserResponse>(this.apiUrl, request);
  }

  getById(id: number): Observable<UserResponse> {
    return this.http.get<UserResponse>(`${this.apiUrl}/${id}`);
  }

  list(): Observable<UserResponse[]> {
    return this.http.get<UserResponse[]>(this.apiUrl);
  }

  getMyProfile(): Observable<UserResponse> {
    return this.http.get<UserResponse>(`${this.apiUrl}/me`);
  }

  updateProfile(request: UpdateProfileRequest): Observable<UserResponse> {
    return this.http.put<UserResponse>(`${this.apiUrl}/me`, request);
  }

  toggleStatus(userId: number): Observable<UserResponse> {
    return this.http.put<UserResponse>(`${this.apiUrl}/${userId}/toggle-status`, {});
  }
}
